### PR TITLE
Add common component

### PR DESCRIPTION
**Pull request by phamilton**
Initial commit of the "common" component of the knative-kafka project as well as top level files (README etc.).  Additional components will be added in separate PRs when the common code has been merged.

### CODE REVIEW

**Review by devguyio — CHANGES_REQUESTED** [review]
Initial check on mandatory files and repo structure as per the [template](https://github.com/kyma-project/community/tree/master/guidelines/repository-guidelines/repository-template) shows two missing files: CODEOWNERS and LICENSE.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+Knative Eventing Contrib Kafka Channel of which you should only install one. 
+
+- [knative-kafka-dispatcher](./components/dispatcher/README.md) - This application runs the Kafka ConsumerGroup 
+responsible for processing messages from the corresponding Kafka Topic.  This is the "Consumer" from the Kafka
+perspective.  The  knative-kafka-controller creates a single instance of this application for each Knative Subscription 
+CustomResource. 
+
+The following diagram shows the message flow through the various components. 
+
+![alt text](./architecture.png "Konduit-Kafka Architecture")
```
Comment (devguyio) [inline README.md]: Maybe remove the reference of `konduit` from the filename and from the diagram?

File: components/common/Makefile [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+# Project Directories
+BUILD_ROOT:=$(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
+BUILD_DIR:=$(BUILD_ROOT)/build
+
+# Application Names (Align With Knative Eventing Version)
+APP_NAME=knative-kafka-common
+APP_VERSION=latest
+
+# Build / Packaging / Deployment Configuration
+APP_PATH=konduit.docker.repositories.sap.ondemand.com/com.sap.c4konduit
```
Comment (devguyio) [inline components/common/Makefile]: probably best to change this URL to a gcs repo of Kyma?

File: components/common/pkg/kafka/constants/constants.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
+package constants
+
+// Constants
+const (
+	// Duration Convenience
+	MillisPerDay = 24 * 60 * 60 * 1000 // 86400000
+
+	// Kafka Secret Label
+	KafkaSecretLabel = "konduit.cx/kafka-secret"
```
Comment (devguyio) [inline components/common/pkg/kafka/constants/constants.go]: as above, a konduit reference

File: components/common/pkg/prometheus/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Prometheus Metrics HTTP Server
+
+This code will provides the ability to start/stop a native Golang HTTP Server backed by the default 
+[Prometheus](https://github.com/prometheus/client_golang) handler.  The desired HTTP Port & Path are
+specified upon server creation.  This should allow the various Konduit "components" to quickly add
```
Comment (devguyio) [inline components/common/pkg/prometheus/README.md]: Konduit reference

**Review by devguyio — CHANGES_REQUESTED** [review]
Second review round shows some names and URLs that need to change

File: README.md (hunk 1 above)
Comment (devguyio) [reply README.md]: also noticed now `CPI`, `IFLOW` and `C4C`

Comment (phamilton) [reply README.md]: For now I'll just remove the diagram, we can redo it later if it makes sense.

File: components/common/Makefile (hunk 2 above)
Comment (phamilton) [reply components/common/Makefile]: Actually for the common component its no longer needed, was only needed to be compatible with SAP build systems.  So I removed it.

File: components/common/pkg/prometheus/README.md (hunk 4 above)
Comment (phamilton) [reply components/common/pkg/prometheus/README.md]: Good catch, updated.

File: components/common/pkg/kafka/constants/constants.go (hunk 3 above)
Comment (phamilton) [reply components/common/pkg/kafka/constants/constants.go]: Good catch.

**Review by devguyio — APPROVED** [review]
lgtm

**Pull request merged**